src/en/darthsdroids/src/eu/kanade/tachiyomi/extension/en/darthsdroids/DarthsDroids.kt

```diff
@@ -0,0 +1,218 @@
+    // cast page. Try to avoid having a character appear more than once
+    // as thumbnail, giving all main characters equal amounts of spotlight.
+    // Pick a character people would intuïtively associate with the
+    // corresponding film, like Qui-Gon for Phantom Menace or Leia for
+    // A New Hope.
+    //
+    // If a book doesn’t have its own cast page, try source a fitting
+    // profile picture from a different page. Avoid sourcing thumbnails
+    // from a different website.
+    private fun dndThumbnailUrlForTitle(nthManga: Int): String = when (nthManga) {
+        // The numbers are assigned in order of appearance of a book on the archive page.
+        0 -> "$baseUrl/cast/QuiGon.jpg" // D&D1
+        1 -> "$baseUrl/cast/Anakin2.jpg" // D&D2
+        2 -> "$baseUrl/cast/ObiWan3.jpg" // D&D3
+        3 -> "$baseUrl/cast/JarJar2.jpg" // JJ
+        4 -> "$baseUrl/cast/Leia4.jpg" // D&D4
+        5 -> "$baseUrl/cast/Han5.jpg" // D&D5
+        6 -> "$baseUrl/cast/Luke6.jpg" // D&D6
+        7 -> "$baseUrl/cast/Cassian.jpg" // R1
+        8 -> "$baseUrl/cast/C3PO4.jpg" // Muppets
+        9 -> "$baseUrl/cast/Finn7.jpg" // D&D7
+        10 -> "$baseUrl/cast/Han4.jpg" // Solo
+        11 -> "$baseUrl/cast/Hux8.jpg" // D&D8
+        // Just some nonsense fallback that screams »Star Wars« but is also so recognisably
+        // OT that one can understand it’s a mere fallback. Better thumbnails require an
+        // extension update.
+        else -> "$baseUrl/cast/Vader4.jpg"
+    }
+
+    private fun dndManga(archiveUrl: String, mangaTitle: String, mangaStatus: Int, nthManga: Int): SManga = SManga.create().apply {
+        setUrlWithoutDomain(archiveUrl)
+        thumbnail_url = dndThumbnailUrlForTitle(nthManga)
+        title = mangaTitle
+        author = "David Morgan-Mar & Co."
+        artist = "David Morgan-Mar & Co."
+        description = """What if Star Wars as we know it didn't exist, but instead the
+            |plot of the movies was being made up on the spot by players of
+            |a Tabletop Game?
+            |
+            |Well, for one, the results might actually make a lot more sense,
+            |from an out-of-story point of view…
+        """.trimMargin()
+        genre = "Campaign Comic, Comedy, Space Opera, Science Fiction"
+        status = mangaStatus
+        update_strategy = when (mangaStatus) {
+            SManga.COMPLETED -> UpdateStrategy.ONLY_FETCH_ONCE
+            else -> UpdateStrategy.ALWAYS_UPDATE
+        }
+        initialized = true
+    }
+
+    override fun popularMangaRequest(page: Int): Request =
+        GET("$baseUrl/archive.html", headers)
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val mainArchive = response.asJsoup()
+        val archiveData = mainArchive.select("div.text > table.text > tbody > tr")
+
+        val mangas = mutableListOf<SManga>()
+        var nextMangaTitle = name
+        var nthManga = 0
+
+        run stop@{
```

Comment:
why `run @stop`?
just loop through it directly

Reply:
a regular for loop can be broken using `break`. or am I missing something?

Reply:
I vaguely remember `return@forEach` not having behaved like `break` but like `continue` in other languages, hence the `run`. I *think* i also tried `break`, but i don’t remember. Kotlin’s weird, coming from other langs. I’ll try the `break`.

Reply:
I might be misremembering then..., should be fine then